http/oidc/src/main/java/org/wildfly/security/http/oidc/JsonWebToken.java

```diff
@@ -297,7 +297,14 @@ public String getPrincipalName(OidcClientConfiguration deployment) {
             case NICKNAME:
                 return getNickName();
             default:
-                return getSubject();
+                String claimValue = getClaimValueAsString(attr);
+                if (claimValue != null) {
+                    return claimValue;
+                } else {
+                    // fall back to sub claim
+                    log.principalAttributeClaimDoesNotExist(attr);
```

Comment:
This looks like it has the potential to become per-request - I am wondering if this should instead be debug information?

Reply:
Good catch, I've fixed this, thanks!